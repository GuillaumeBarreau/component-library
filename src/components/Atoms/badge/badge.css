/* badge */
.badge {
  margin: 0.25rem;
  border-radius: 1rem;
  padding: 0.4rem 0.8rem;
  display: flex;
  align-items: center;
  letter-spacing: 0.0625rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.badge > svg {
  margin-right: 0.6rem;
}

.badge > p {
  text-transform: uppercase;
}

.badge--notice-default {
  color: var(--color-notice-default--text);
  background-color: var(--color-notice-default--background);
  border: 2px solid var(--color-base-02--light);
}

.badge--notice-error {
  color: var(--color-notice-error--text);
  background-color: var(--color-notice-error--background);
  border: 2px solid var(--color-notice-error--default);
}

.badge--notice-success {
  color: var(--color-notice-success--text);
  background-color: var(--color-notice-success--background);
  border: 2px solid var(--color-notice-success--default);
}

.badge--notice-warning {
  color: var(--color-notice-warning--text);
  background-color: var(--color-notice-warning--background);
  border: 2px solid var(--color-notice-warning--default);
}

.badge--notice-info {
  color: var(--color-notice-info--text);
  background-color: var(--color-notice-info--background);
  border: 2px solid var(--color-notice-info--default);
}

.badge--override-typography {
  margin: 0;
}
