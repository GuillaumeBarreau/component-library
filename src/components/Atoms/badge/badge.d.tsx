import { TBadgeIconName } from "components/_settings/globalTypes/Types.d";

export interface ComponentBadgeProps {
  children: string;
  notice?: "default" | "success" | "error" | "info" | "warning";
  className?: string;
  size?: "xsmall" | "small" | "default";
  iconName?: TBadgeIconName;
}
