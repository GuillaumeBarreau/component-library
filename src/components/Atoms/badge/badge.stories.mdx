import { Meta, Story, Canvas, ArgsTable } from "@storybook/addon-docs";
import { Badge } from "./badge";

<Meta 
  title="Components/Atoms/Badge" 
  component={Badge}
  argTypes={{
  }}/>

# Badge

The Badge component is used to quickly provide additional information on an item (such as a number of elements, a category, a indication...).

## React Component

#### Import library

```
Import {Atoms} from '@bonjour-les-pixels/component-library';
```

#### Use Badge component

```
<Atoms.Badge> </Atoms.Badge>
```

<Canvas>
  <Badge
    children="badge"
    notice="info"
  />
</Canvas>

### Props

<ArgsTable story="." />

### Notice

You can use the `notice` prop to modify the badge color. This props can be declined in 5 ways. The style varies according to the level of importance of the action in the page context.

<Canvas>
    <Badge
      handleClick={() => console.log("here")}
      children="Default"
    />
    <Badge
      notice="success"
      children="success"
    />
    <Badge
      notice="info"
      children="info"
    />
    <Badge
      notice="error"
      children="error"
    />
    <Badge
      notice="warning"
      children="warning"
    />
</Canvas>

### ClassName

You can use the `className` prop to override a component's styles using custom classes.

```
  <Badge
  className="override-style-componenent"
/>
```

### IconName

You can use the iconName prop to have icons. You will find a list icons available below. 

<Canvas>
  <Badge
    notice="success"
    iconName="circle-exclamation"
    children="button"
  />
</Canvas>

### Size

You can use the size prop to modify the button size. You can set the value to `xsmal`, `small` or `default`. 
The default size is default and therefore doesn't need to be specified.

<Canvas>
  <Badge
      size="xsmall"
      children="Xsmall"
      notice="info"
    />
  <Badge
      size="small"
      children="Small"
      notice="info"
    />
  <Badge
      size="default"
      children="Default"
      notice="info"
    />
  <Badge
      size="medium"
      children="Medium"
      notice="info"
    />
</Canvas>