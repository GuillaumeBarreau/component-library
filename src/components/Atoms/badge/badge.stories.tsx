import { Badge } from "./badge";
import { Meta } from "@storybook/react/types-6-0";
import { ComponentBadgeProps } from "./badge.d";

export default {
  title: "Badge",
  component: Badge,
  parameters: {
    a11y: {
      // optional selector which element to inspect
      element: "#root",
      // optional flag to prevent the automatic check
      manual: false,
    },
  },
} as Meta;

const Template: any = (args: JSX.IntrinsicAttributes & ComponentBadgeProps) => (
  <>
    <section style={{ display: "flex", justifyContent: "center" }}>
      <Badge {...args} />
    </section>
  </>
);

export const BadgeDefault = Template.bind({});
BadgeDefault.args = {
  children: "badge",
};
