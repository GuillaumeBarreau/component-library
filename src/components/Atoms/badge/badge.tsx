import React from "react";
import classNames from "classnames";
import { ComponentBadgeProps } from "./badge.d";
import { Typography } from "components/Atoms/typography";
import { Icon } from "components/Atoms/icon";
import "./badge.css";

export const Badge: React.FC<ComponentBadgeProps> = (props) => {
  const {
    children,
    className,
    notice = "default",
    size = "default",
    iconName,
  } = props;

  const prefixCls: string = "badge";

  const classes = classNames(
    prefixCls,
    {
      [`${prefixCls}--notice-${notice}`]: notice,
    },
    className
  );

  const iconNameSize =
    size === "xsmall" ? "2xs" : size === "small" ? "2xs" : "sm";

  const iconNode: JSX.Element | null =
    iconName && iconNameSize ? (
      <Icon icon={iconName} size={iconNameSize} />
    ) : null;

  return (
    <div className={classes}>
      {iconNode}
      <Typography
        font="bold"
        size={size}
        variant={"p"}
        className="badge--override-typography"
      >
        {children}
      </Typography>
    </div>
  );
};
