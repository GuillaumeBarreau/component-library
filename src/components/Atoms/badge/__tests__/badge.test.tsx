import React from "react";
import { shallow } from "enzyme";
import { Badge } from "../badge";
import { ComponentBadgeProps } from "../badge.d";

describe("Badge", () => {
  let defaultProps: JSX.IntrinsicAttributes & ComponentBadgeProps;

  beforeEach(() => {
    defaultProps = {
      children: "Badge",
      className: "overide-class",
    };
  });

  it("Should match with Snapshot correctly", () => {
    const wrapper = shallow(<Badge {...defaultProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("mount correctly", () => {
    const wrapper = shallow(<Badge {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
  });

  it("should render with notice props", () => {
    const props: JSX.IntrinsicAttributes & ComponentBadgeProps = {
      ...defaultProps,
      notice: "success",
    };
    const wrapper = shallow(<Badge {...props} />);
    expect(wrapper.find(".badge--notice-success")).toHaveLength(1);
  });

  it("should render with className props", () => {
    const props: JSX.IntrinsicAttributes & ComponentBadgeProps = {
      ...defaultProps,
      className: "overide-class",
    };
    const wrapper = shallow(<Badge {...props} />);
    expect(wrapper.find(".overide-class")).toHaveLength(1);
  });

  it("should render with size props", () => {
    const props: JSX.IntrinsicAttributes & ComponentBadgeProps = {
      ...defaultProps,
      size: "small",
    };
    const wrapper = shallow(<Badge {...props} />);
    expect(wrapper.find(".typography-small")).toHaveLength(1);
  });

  it("should render with children props", () => {
    const props: JSX.IntrinsicAttributes & ComponentBadgeProps = {
      ...defaultProps,
      children: "Badge",
    };
    const wrapper = shallow(<Badge {...props} />);
    expect(wrapper.html()).toContain("Badge");
  });
});
