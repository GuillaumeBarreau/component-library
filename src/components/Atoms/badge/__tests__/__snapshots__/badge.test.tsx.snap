// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Should match with Snapshot correctly 1`] = `
"<div className=\\"badge badge--notice-default overide-class\\">
  <Typography font=\\"bold\\" size=\\"default\\" variant=\\"p\\" className=\\"badge--override-typography\\">
    Badge
  </Typography>
</div>"
`;
